stl/inc/future

```diff
@@ -1319,9 +1346,9 @@ public:
     }
 
     void reset() { // reset to newly constructed state
-        _MyStateManagerType& _State = _MyPromise._Get_state_for_set();
-        _MyStateType* _MyState      = static_cast<_MyStateType*>(_State._Ptr());
-        _MyPromiseType _New_promise(new _MyStateType(_MyState->_Get_fn()));
+        _MyStateManagerType& _State_mgr = _MyPromise._Get_state_for_set();
+        _MyStateType& _MyState          = *static_cast<_MyStateType*>(_State_mgr._Ptr());
+        _MyPromiseType _New_promise(new _MyStateType(_STD move(_MyState)._Get_fn()));

```

Comment:
It's perhaps necessary to `move` per [[futures.task.members]/26](https://eel.is/c++draft/futures.task.members#26). However, the behavior of formerly valid code will change. Not sure whether this is acceptable.

Reply:
It seems possible to keep old behavior for existing types while enable correct behavior for newly allowed move-only functors - make the new branch of `_Copy` do `const_cast` and then call `_Move`. But that's somehow weird.

Reply:
This uniform approach is reasonable.